src/stores/QtumPredictionStore.js

```diff
@@ -0,0 +1,72 @@
+import { observable, action, runInAction, reaction } from 'mobx';
+import _ from 'lodash';
+import { Token, OracleStatus, AppLocation } from '../constants';
+import { queryAllOracles } from '../network/graphQuery';
+import Oracle from './models/Oracle';
+
+
+export default class AllPredictionsStore {
+  @observable loading = true
+  @observable loadingMore = false
+  @observable list = []
+  @observable hasMore = true
+  @observable skip = 0
+  limit = 50
+
+  constructor(app) {
+    this.app = app;
+    reaction(
+      () => this.app.sortBy, // when 'sortBy' changes
+      () => {
+        // and we're on the AllEvents page
+        if (this.app.ui.location === AppLocation.qtumPrediction) {
+          this.init(this.skip); // fetch new events
+        }
+      }
+    );
+  }
+
+  @action.bound
+  async init(limit = this.limit) {
+    if (limit === this.limit) {
+      this.skip = 0;
+    }
+    this.hasMore = true;
+    this.app.ui.location = AppLocation.qtumPrediction;
+    this.list = await this.fetchQtumPredictions(limit);
+    runInAction(() => {
+      this.skip += this.limit;
```

Comment:
shouldn't it use the local var `limit` instead of `this.limit` since if its passed in as a param it might be different?